### PR TITLE
Updated LAMMPS package to use gpu and kokkos options.

### PR DESCRIPTION
**Pull request by frobnitzem**

### CODE REVIEW

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -49,17 +49,23 @@ def url_for_version(self, version):
                           'coreshell', 'dipole', 'granular', 'kspace', 'latte',
                           'manybody', 'mc', 'meam', 'misc', 'molecule',
                           'mpiio', 'peri', 'poems', 'python', 'qeq', 'reax',
-                          'replica', 'rigid', 'shock', 'snap', 'srd',
+                          'replica', 'rigid', 'shock', 'snap', 'spin', 'srd',
                           'user-atc', 'user-h5md', 'user-lb', 'user-misc',
                           'user-netcdf', 'user-omp', 'voronoi']
+    sm_levels = ('sm_20', 'sm_30', 'sm_50', 'sm_60', 'sm_70', 'sm_75')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lammps/package.py]: Might want to add the `CudaPackage` base class to be consistent with other packages?

Comment (frobnitzem) [reply var/spack/repos/builtin/packages/lammps/package.py]: The class now extends `CudaPackage` and incorporates recent updates to the package.

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 2]
```diff
@@ -125,6 +135,17 @@ def cmake_args(self):
...
+            args.append('-DPKG_GPU=ON')
+            args.append('-DGPU_API=cuda')
+            cuda_arch = spec.variants['cuda_arch'].value
+            if cuda_arch is not None:
+                args.append('-DGPU_ARCH=sm_{0}'.format(cuda_arch[0]))
+            args.append('-DCUDA_MPS_SUPPORT={0}'.format(
+                'ON' if '+cuda_mps' in spec else 'OFF'))
+        elif spec.satisfies('+opencl'):
+            args.append('-DPKG_GPU=ON')
+            args.append('-DGPU_API=opencl')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lammps/package.py]: Probably also want:
```python
else:
    args.append('-DPKG_GPU=OFF')
```

**Comment by adamjstewart** [comment]
@alalazo I forget if you care about LAMMPS or not. If you do, you might want to review this PR.

**Comment by alalazo** [comment]
> I forget if you care about LAMMPS or not. If you do, you might want to review this PR.

@adamjstewart Not specifically. Anything that is good for you in this PR is also good for me.

File: var/spack/repos/builtin/packages/lammps/package.py (hunk 2 above)
Comment (frobnitzem) [reply var/spack/repos/builtin/packages/lammps/package.py]: GPU is off by default, but I'll add it.

**Comment by adamjstewart** [comment]
It looks like you had a bad merge, and several commits you didn't want have been added to the PR. If you're note super familiar with git, it might just be easiest to make a new PR with only the changes you want.

**Pull request closed without merging**